gateway/api/views.py

```diff
@@ -0,0 +1,249 @@
+"""Views."""
+
+import json
+import os.path
+import shutil
+import tarfile
+import uuid
+
+import requests
+from allauth.socialaccount.providers.keycloak.views import KeycloakOAuth2Adapter
+from dj_rest_auth.registration.views import SocialLoginView
+from django.conf import settings
+from django.contrib.auth.models import User  # pylint: disable=imported-auth-user
+from ray.dashboard.modules.job.common import JobStatus
+from ray.dashboard.modules.job.sdk import JobSubmissionClient
+from rest_framework import viewsets, permissions, status
+from rest_framework.decorators import action
+from rest_framework.generics import get_object_or_404
+from rest_framework.response import Response
+from rest_framework.views import APIView
+
+from .models import Program, Job, ComputeResource
+from .permissions import IsOwner
+from .serializers import ProgramSerializer, JobSerializer
+from .utils import try_json_loads
+
+
+# pylint: disable=too-many-ancestors
+class ProgramViewSet(viewsets.ModelViewSet):
+    """ProgramViewSet."""
+
+    queryset = Program.objects.all()
+    serializer_class = ProgramSerializer
+    permission_classes = [permissions.IsAuthenticated]
+
+    def get_queryset(self):
+        return Program.objects.all().filter(author=self.request.user)
+
+    def perform_create(self, serializer):
+        serializer.save(author=self.request.user)
+
+    @action(methods=["POST"], detail=False)
+    def run_program(self, request):
+        """Runs provided program on compute resources."""
+
+        serializer = self.get_serializer(data=request.data)
+        if serializer.is_valid():
+            # create program
+            program = Program(**serializer.data)
+            _, dependencies = try_json_loads(program.dependencies)
+            _, arguments = try_json_loads(program.arguments)
```

Comment:
Can this be solved with the [validators](https://github.com/Qiskit-Extensions/quantum-serverless/tree/main/repository/api/validators)?

Reply:
Let's add validation as a follow up PR